packages/clerk-js/src/utils/injectedWeb3Providers.ts

```diff
@@ -53,13 +53,15 @@ class InjectedWeb3Providers {
   }
 
   get = (provider: InjectedWeb3Provider) => {
-    // In case there is a single injected provider, use that directly.
-    // This is needed in order to support other compatible Web3 Wallets (e.g. Rabby Wallet)
-    // and maintain the previous behavior
-    if (this.#providers.length === 1) {
-      return this.#providers[0].provider;
+    const ethProvider = this.#providers.find(p => p.info.name === this.#providerIdMap[provider])?.provider;
+    if (ethProvider !== undefined) {
+      return ethProvider;
     }
-    return this.#providers.find(p => p.info.name === this.#providerIdMap[provider])?.provider;
+
+    // In case we weren't able to find the requested provider, fallback to the
+    // global injected provider instead, if any, to allow the user to continue
+    // the flow rather than be a no-op
+    return window.ethereum;
```

Comment:
1. Maybe i'm mistaken, but i think we were doing this and we removed it.
1. Is there a scenario where if want to request with metamask but the browser does not have it i should throw an error instead ? If yes, maybe the fallback needs to happen at the component level.

Reply:
1. That's not what we were doing before. We were explicitly trying to find the requested provider (e.g. MetaMask, OKX Wallet etc) and if we couldn't, we weren't initiated the flow, giving the user the impression that the application is broken
2. Maybe we are looking to throw an error if there isn't a web3 provider available, but we are looking to investigate that in a different effort